### PR TITLE
Feature/use postgre sql

### PR DESCRIPTION
**Pull request by 0xLeif**

### CODE REVIEW

**Comment by 0xLeif** [comment]
Linux Tests:
```
Test Suite 'ValidatorServiceTests' passed at 2020-02-27 00:56:03.572
	 Executed 5 tests, with 0 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'debug.xctest' passed at 2020-02-27 00:56:03.572
	 Executed 14 tests, with 0 failures (0 unexpected) in 1.996 (1.996) seconds
Test Suite 'All tests' passed at 2020-02-27 00:56:03.572
	 Executed 14 tests, with 0 failures (0 unexpected) in 1.996 (1.996) seconds
```

**Comment by 0xLeif** [comment]
Ran Tests with: `swift test --enable-test-discovery`

**Comment by 0xLeif** [comment]
Fixed Tests so one doesn't have to use `swift test --enable-test-discovery`.

One can simply use `swift test`

**Comment by 0xLeif** [comment]
Seems like tests are failing due to the fact that the CI doesn't have a PostgreSQL server running locally...

File: Sources/App/configure.swift [hunk 1]
```diff
@@ -29,13 +29,16 @@ public func configure(_ config: inout Config, _ env: inout Environment, _ servic
     services.register(middlewares)
     
     config.prefer(MemoryKeyedCache.self, for: KeyedCache.self)
-    // Configure a SQLite database
-    let sqlite = try SQLiteDatabase(storage: .memory)
+
+    // Configure a PostgreSQL database
+    let config = PostgreSQLDatabaseConfig(hostname: "localhost", port: 5432, username: "leif", database: "leif", password: "sudo", transport: .cleartext)
```
Comment (0xLeif) [inline Sources/App/configure.swift]: Need to change to use a default one for DEBUG and one for RELEASE

**Review by ambid17 — APPROVED** [review]

File: Sources/App/configure.swift [hunk 2]
```diff
@@ -29,13 +29,17 @@ public func configure(_ config: inout Config, _ env: inout Environment, _ servic
     services.register(middlewares)
     
     config.prefer(MemoryKeyedCache.self, for: KeyedCache.self)
-    // Configure a SQLite database
-    let sqlite = try SQLiteDatabase(storage: .memory)
+
+    // Configure a PostgreSQL database
+    let config = PostgreSQLDatabaseConfig(hostname: "localhost", port: 5432, username: "oneleif", database: "website", password: "sudo", transport: .cleartext)
```
Comment (ambid17) [inline Sources/App/configure.swift]: dont need password

File: Sources/App/configure.swift [hunk 3]
```diff
@@ -29,13 +29,17 @@ public func configure(_ config: inout Config, _ env: inout Environment, _ servic
...
+
+    let postgres = PostgreSQLDatabase(config: config)
     
-    // Register the configured SQLite database to the database config.
+    // Register the configured PostgreSQL database to the database config.
     var databases = DatabasesConfig()
-    databases.add(database: sqlite, as: .sqlite)
```
Comment (ambid17) [inline Sources/App/configure.swift:37]: check out multiline comments

**Comment by arkus7** [comment]
I'll review the code in my evening 😛I can manage the circleci to create database while testing 😀

File: Sources/App/configure.swift (hunk 2 above)
Comment (arkus7) [inline Sources/App/configure.swift]: I think we should load the credentials from environment variables, so we can quickly change the credentials without having to update the application code.

```swift
let config = PostgreSQLDatabaseConfig(
  hostname: Environment.get("DB_HOST") ?? "localhost", 
  port: Int(Environment.get("DB_PORT") ?? "") ?? 5432, 
  username: Environment.get("DB_USER") ?? "oneleif", 
  database: Environment.get("DB_NAME") ?? "oneleif-\(env.name)", 
  password: Environment.get("DB_PASSWORD") ?? "root", 
  transport: .cleartext
)
```

**Review by arkus7 — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 4]
```diff
@@ -8,6 +8,11 @@ jobs:
   test:
     docker:
       - image: swift:4.1
+      - image: circleci/postgres:latest
```
Comment (arkus7) [inline .circleci/config.yml:11]: We need this also for the `test-release` job

**Review by arkus7 — CHANGES_REQUESTED** [review]

File: Sources/App/configure.swift [hunk 5]
```diff
@@ -29,13 +29,23 @@ public func configure(_ config: inout Config, _ env: inout Environment, _ servic
     services.register(middlewares)
     
     config.prefer(MemoryKeyedCache.self, for: KeyedCache.self)
-    // Configure a SQLite database
-    let sqlite = try SQLiteDatabase(storage: .memory)
+
+    // Configure a PostgreSQL database
+    let config = PostgreSQLDatabaseConfig(hostname: "localhost", port: 5432, username: "postgres", database: "website", transport: .cleartext)
```
Comment (arkus7) [inline Sources/App/configure.swift]: Remove this, the config is duplicated now

**Review by arkus7 — CHANGES_REQUESTED** [review]

**Review by arkus7 — APPROVED** [review]

**Pull request merged**